### PR TITLE
Add tests to remaining EngagementCoordinator events

### PR DESCRIPTION
**Pull request by gersonnoboa**
Tests for secure conversations and calls have been added. There are some parts that I have left out because they are unnecessarily hard to test, and others require me opening up private properties.

MOB-2919

**Release notes:**

 - [ ] Feature
 - [x] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

### CODE REVIEW

File: GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+extension EngagementCoordinatorTests {
+    // Start
+    func test_call() {
+        coordinator = createCoordinator(withKind: .audioCall)
+        coordinator.start()
+        
+        let viewController = coordinator.navigationPresenter.viewControllers.first as? CallViewController
+
+
```
Comment (yurii-glia) [inline GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift:13]:
```suggestion
```

File: GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        XCTAssertEqual(coordinator.interactor.state, .enqueueing(.audio))
+    }
+
+    func test_video() {
+        coordinator = createCoordinator(withKind: .videoCall)
+        coordinator.start()
+
+        let viewController = coordinator.navigationPresenter.viewControllers.first as? CallViewController
+
+
```
Comment (yurii-glia) [inline GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift:24]:
```suggestion
```

File: GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift [hunk 3]
```diff
@@ -0,0 +1,136 @@
+import Foundation
+import XCTest
+@testable import GliaWidgets
+
+extension EngagementCoordinatorTests {
+    // Start
```
Comment (yurii-glia) [inline GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift:6]: if we would like to separate tests by sections, would it make sense to use pragma marks?
```
// MARK: - Start
```

File: GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        coordinator = createCoordinator(withKind: .videoCall)
+        var calledEvents: [EngagementCoordinator.DelegateEvent] = []
+
+        coordinator.delegate = { event in
+            calledEvents.append(event)
+        }
+
+        coordinator.start()
+
+        let callCoordinator = coordinator.coordinators.first {$0 is CallCoordinator} as? CallCoordinator
```
Comment (yurii-glia) [inline GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift:41]:
```suggestion
        let callCoordinator = coordinator.coordinators.first { $0 is CallCoordinator } as? CallCoordinator
```

File: GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+        coordinator.delegate = { event in
+            calledEvents.append(event)
+        }
+
+        coordinator.start()
+
+        let chatCoordinator = coordinator.coordinators.first { $0 is ChatCoordinator } as? ChatCoordinator
+
+        let mediaUpgradeOffer = try! CoreSdkClient.MediaUpgradeOffer(type: .audio, direction: .twoWay)
```
Comment (yurii-glia) [inline GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift:60]:
```suggestion
        let mediaUpgradeOffer = try CoreSdkClient.MediaUpgradeOffer(type: .audio, direction: .twoWay)
```

File: GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        chatCoordinator?.delegate?(
+            .mediaUpgradeAccepted(
+                offer: mediaUpgradeOffer,
+                answer: { answer, success in success?(true, nil) }
+            )
+        )
+
+        let callCoordinator = coordinator.coordinators.first {$0 is CallCoordinator} as? CallCoordinator
```
Comment (yurii-glia) [inline GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift:68]:
```suggestion
        let callCoordinator = coordinator.coordinators.first { $0 is CallCoordinator } as? CallCoordinator
```

File: GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+        coordinator = createCoordinator(withKind: .videoCall)
+        var calledEvents: [EngagementCoordinator.DelegateEvent] = []
+
+        coordinator.delegate = { event in
+            calledEvents.append(event)
+        }
+
+        coordinator.start()
+
+        let callCoordinator = coordinator.coordinators.first {$0 is CallCoordinator} as? CallCoordinator
```
Comment (yurii-glia) [inline GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift:105]:
```suggestion
        let callCoordinator = coordinator.coordinators.first { $0 is CallCoordinator } as? CallCoordinator
```

File: GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+        callCoordinator?.delegate?(.minimize)
+
+        callAfterTimeout {
+            XCTAssertEqual(calledEvents.last, .minimized)
+        }
+    }
+
+//    func test_callDelegateFinished() throws {
```
Comment (yurii-glia) [inline GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift:113]: will we use this test in future?

File: GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+        }
+    }
+
+    func test_callDelegateEngaged() throws {
+        coordinator = createCoordinator(withKind: .videoCall)
+        
+        coordinator.start()
+
+        try showGliaViewController()
+        let callCoordinator = coordinator.coordinators.first { $0 is CallCoordinator } as! CallCoordinator
```
Comment (igorkravchenko) [inline GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift:84]:
```suggestion
        let callCoordinator =  try XCTUnwrap(coordinator.coordinators.first { $0 is CallCoordinator } as? CallCoordinator)
```

**Review by igorkravchenko — COMMENTED** [review]
I think force unwrapping can be replaced with `XCTUnwrap`  to avoid crashes.

File: GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift (hunk 8 above)
Comment (gersonnoboa) [reply GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift]: Nope, fixed bbb78205f0b4d4a0c13507fd9f24409ed42d3523

File: GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift (hunk 3 above)
Comment (gersonnoboa) [reply GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift]: Fixed 6964ad8bae2e13f49b75f2e6cfee601c7ca2c4a1

**Review by igorkravchenko — APPROVED** [review]

**Review by yurii-glia — APPROVED** [review]

**Comment by gersonnoboa** [comment]
!squash

**Pull request merged**